tests/saitests/sai_qos_tests.py

```diff
@@ -60,6 +61,41 @@
 ECN_INDEX_IN_HEADER = 53 # Fits the ptf hex_dump_buffer() parse function
 DSCP_INDEX_IN_HEADER = 52 # Fits the ptf hex_dump_buffer() parse function
 
+def get_rx_port(dp, device_number, src_port_id, dst_mac, dst_ip, src_ip):
+    ip_id = 0xBABE
+    tos = (0 << 2) | 1
+    src_port_mac = dp.dataplane.get_mac(device_number, src_port_id)
+    pkt = testutils.simple_ip_packet(pktlen=64,
+                            eth_dst=dst_mac,
+                            eth_src=src_port_mac,
+                            ip_src=src_ip,
+                            ip_dst=dst_ip,
+                            ip_tos=tos,
+                            ip_id=ip_id)
+
+    send_packet(dp, src_port_id, pkt, 1)
+
+    exp_pkt = testutils.simple_ip_packet(pktlen=48,
```

Comment:
why pktlen is 48 rather than 64 (line 68)

Reply:
only part of the 48 bytes are checked for the expected packet. Other fields, like the payload are ignored. It can be less than 48 too.

Reply:
What are the fields used for hashing across lags? Are they going to remain the same for all the packets sent?

Reply:
2 hdrs starting from forwarding header, so ASICs use 5 tuple in case of UDP or TCP. Otherwise 3 parameters. As long as test API and get_rx_port() use the same API to generate packet, the packets will hash to the same port.